Edit form db

Liliya-V opened the pull request:
Editing form is now working - i can now data, and it will be changes at the DB as well as on a Portfolio page.

Liliya-V committed:
created editformdb.php, wrote a code for inserting editeddata into a DB

Liliya-V committed:
changes a title

Liliya-V committed:
Merge branch 'master' into editformdb

Liliya-V committed:
added pictures

Liliya-V committed:
merged master branch in.
set an id to edit form sochanged a select insert function to update one, set an id to edit a correct position

Liliya-V committed:
set a substraction of first 7 characters from an 'image' field of an edit form, so the path to the image would not have a doubled directory

mporam commented on adminPanel/editform.php:
```diff
@@ -0,0 +1,35 @@
+$db = new PDO("mysql:host=192.168.20.20;dbname=PortfolioProjects", 'root', '');
+$db -> setAttribute(PDO::ATTR_DEFAULT_FETCH_MODE, PDO::FETCH_ASSOC);
+$sql = "SELECT `title`, `link`, `image` FROM `projects` WHERE `id` = :id;";
+$id = $_GET['id'];
```
no check if the get id is empty

mporam commented on adminPanel/editform.php:
```diff
@@ -0,0 +1,35 @@
+        <form  action="editformdb.php?id=<?php echo $id;?>" method="POST">
+            <div class="container">
+                <?php
+                    echo '<label for="title" > Type a title here:</label >';
```
you can do this much tidier but just wrapping the php in php tags, like:
```
<label for="title" > Type a title here:</label >
<input class="addform" type = "text" name = "title" value="<?php echo htmlspecialchars($project['title']);?>" >
```

mporam commented on adminPanel/editformdb.php:
```diff
@@ -0,0 +1,26 @@
+require('functions.php');
+$db = new PDO("mysql:host=192.168.20.20;dbname=PortfolioProjects", 'root', '');
+$db->setAttribute(PDO::ATTR_DEFAULT_FETCH_MODE, PDO::FETCH_ASSOC);
+$id = $_GET['id'];
```
no check if id is empty

mporam commented on adminPanel/editformdb.php:
```diff
@@ -0,0 +1,26 @@
+    $data = setVariables($_POST);
+    $data[':id'] = $id;
+    $addProject = $queryProjects->execute($data);
+    echo 'Project was successfully edited.';
```
you dont know this worked successfully. you need to do an if on `$addProject` to see if the query worked

Liliya-V committed:
Added if statement to check if an UPDATE query was executed successfukly,
Tityed the code by moving PHP echoes inside of the inputs,
Added an if statement to check if a GET 'id' is empty or not.

mporam commented on adminPanel/editformdb.php:
```diff
@@ -0,0 +1,22 @@
+            echo 'Project was successfully edited.';
+            echo '<a href="../portfolio.php">Go to your portfolio. </a>';
+        }
+}
```
need to check thedata is valid or show an error

Liliya-V committed:
added ELSE statement for if the functuon wouldn't work

Liliya-V committed:
Merge branch 'master' into edit-form-db

mporam commented on adminPanel/editform.php:
```diff
@@ -2,7 +2,9 @@
 $db -> setAttribute(PDO::ATTR_DEFAULT_FETCH_MODE, PDO::FETCH_ASSOC);
 $sql = "SELECT `title`, `link`, `image` FROM `projects` WHERE `id` = :id;";
-$id = $_GET['id'];
+if (!empty($_GET['id'])) {
```
you need to put the prepare and execute inside this if as well, otherwise it will error.

probably add an else that redirects to the admin panel

Liliya-V committed:
put a prepare-execute commands inside the if statement, added else with redirect to an adminpanel

Liliya-V committed:
Merge branch 'edit-form-db' of github.com:Liliya-V/Portfolio into edit-form-db
